### ISSUE TITLE
Warn when used in create-react-app without macro

### ISSUE THREAD

**Issue by mxstbr** [styled-components/styled-components#2343]
We should figure out whether it's possible to warn styled-components users in dev if they're using the standard library in create-react-app, recommending the use of the Babel macro. It makes the experience so much better thanks to the human-readable classes.

/cc @gaearon who might have an idea how we can detect this, or maybe @kentcdodds is there a way to detect if an app has babel-plugin-macros installed?

**Comment by kentcdodds** [comment]
Considering the fact that styled-components code only runs in the browser I think there wouldn't be a way to detect if babel-plugin-macros is installed. Would be pretty sweet if you could though! Maybe the development version of styled-components could attempt to import the package.json of the project (with a generous try/catch block)?

Even easier:

```javascript
let canUseMacros = false
try {
  require('babel-plugin-macros/package.json')
  canUseMacros = true
} catch (e) {
  // ignore
}
```

Hmmm... Maybe that'd work 🤔

**Comment by mackness** [comment]
@mxstbr Hi Max, is this something that still needs to be worked on? If it is I would not mind picking up this issue :)

**Comment by BenLorantfy** [comment]
I gave this a go and I had trouble determining at runtime if the import is from `/macro` or not. The babel macro re-writes the `styled-components/macro` imports to `styled-components` at compile time. This means at run-time I couldn't tell which import was used.

I'm thinking we could change the babel macro to re-write to something else, say `styled-components/macro-entry`. Then we could tell if the macro was being used. That would mean we would need a change to the rollup config though, and at that point I started wondering if it was overkill.

Another issue I thought of: how do treat cases when styled-components is a transitive dependency? That is, the developer's app depends on a library that depends on styled-components? Do we still want to show the warning if the library does not import from `/macro`? If not, how do we tell where the import is coming from at runtime? 🤔
